### DISCUSSION TITLE
PaddleOCR/ppocr/infer_deploy/cpp_infer

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleOCR#13770] [Q&A]
# PaddleOCR/ppocr/infer_deploy/cpp_infer

Awesome multilingual OCR toolkits based on PaddlePaddle (practical ultra lightweight OCR system, support 80+ languages recognition, provide data annotation and synthesis tools, support training and deployment among server, mobile, embedded and IoT devices)

https://paddlepaddle.github.io/PaddleOCR/ppocr/infer_deploy/cpp_infer.html

**Comment by wangzdev** [comment]
tools/build.sh 脚本在哪了 文档能写成这样吗 根本没法用

**Reply by GreatV** [reply]
https://github.com/PaddlePaddle/PaddleOCR/tree/main/deploy/cpp_infer

**Reply by GreatV** [reply]
![image](https://github.com/user-attachments/assets/7323d2d3-6f70-4dca-90d1-46bfdf926a35)

**Reply by wangzdev** [reply]
感谢，找到了
首先说下，感谢开源，百度这款ocr 还是非常强大的，准确率比手动训练基于 ctc loss function 的模型高多了，解决了实际问题。

我是用于 nvidia jetson xavier  c++ 集成的，目前我个人在 该设备上使用该demo 成功，并使用 tensorrt ，
但变异过程也遇到一些问题，但解决了，分享下，也许对大家有用

1. 首先mkl  要关闭，因为 jetson 使用 openblas
2.编译过程中会遇到 cmake/ppocr.d ....  什么文件找不到 ，查了下 ，是因为 cxx flag 里面有个 -o3 导致，去掉就行
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g  ${FLAG_OPENMP} -std=c++11")
在这 ，那个我已经去掉了
这可能和我cmake 版本较高有关 ，我用的是 camke 3.25
3.如果你是像我一样直接下载了 编译好的 paddle jetson 版本，会发现demo  里有 config.DIsableMKL 这个类似函数找不到，这个我的解决办法是 把这个函数调用给注释掉了
